AirCasting/SearchAndFollow/SearchLocation/SearchViewModel.swift

```diff
@@ -56,11 +56,15 @@ class SearchViewModel: ObservableObject {
     }
     
     func onSensorTap(with name: String) {
-        self.sensorTypes.forEach({ $0.isSelected = false })
+        deselectAll(sensorTypes)
         self.sensorTypes.first(where: { $0.name == name })?.isSelected = true
         self.objectWillChange.send()
     }
     
+    func deselectAll(_ all: [SearchParameter]) {
```

Comment:
<img width="1114" alt="Zrzut ekranu 2022-08-8 o 10 47 38" src="https://user-images.githubusercontent.com/44276070/183378308-96b7a761-c722-4f5c-8b0c-8be8627b22bd.png">
I think I cannot, is there a way to use/test private functions?

Reply:
I can see that there is a way, but it could be a bit overkill in this case I suppose. Should I consider not testing this part of the code and moving it into private? or testing and applying something like this solution from the link? https://www.avanderlee.com/swift/testing-private-methods-variables/

Reply:
I don't have a full context, but can't you just check if this deselection happens when you call `onParameterTap` and `onSensorTap` since those are public methods that are supposed to do the deselection and keep this deselect func private?